lib/simp/rake/pupmod/helpers.rb

```diff
@@ -287,5 +289,96 @@ def changelog_annotation( quiet = false, file = nil )
         Rake::Task[:spec_clean].invoke
       end
     end
+
+    # This hidden task provides a way to create and use a fixtures.yml file
+    # based on an externally specified Puppetfile
+    #
+    # Downloaded repos that do not contain a metadata.json will be removed
+    #
+    # Set the environment variable SIMP_RSPEC_PUPPETFILE to point to a remote Puppetfile
+    #
+    # Set the environment variable SIMP_RSPEC_MODULEPATH to symlink named
+    # modules from the designated directory instead of downloading them.
+    task :custom_fixtures_hook do
+      # Don't do anything if the user has already set a path to their fixtures
+      unless ENV['FIXTURES_YML']
+        short_name = metadata['name'].split('-').last
+
+        fixtures_hash = {
+          'fixtures' => {
+            'symlinks' => {
+              short_name => '#{source_dir}'
+            }
+          }
+        }
+
+        modulepath = ENV['SIMP_RSPEC_MODULEPATH']
+        local_modules = {}
+
+        if modulepath
+          unless File.directory?(modulepath)
+            fail("Could not find a module directory at #{modulepath}")
+          end
+
+          # Grab all of the local modules and convert them into something
+          # that can be turned into a Hash easily
+          local_modules = Hash[Dir.glob(File.join(modulepath, '*', 'metadata.json')).map do |m|
+            [File.basename(File.dirname(m)), File.absolute_path(File.dirname(m))]
+          end]
+
+          local_modules.delete(short_name)
+        end
+
+        puppetfile = ENV['SIMP_RSPEC_PUPPETFILE']
+
+        if puppetfile
+          fail("Could not find Puppetfile at #{puppetfile}") unless File.exist?(puppetfile)
+
+          require 'simp/rake/build/deps'
+
+          puppetfile = R10KHelper.new(puppetfile)
+
+          puppetfile.modules.each do |pupmod|
+            next unless pupmod[:name]
```

Comment:
Only want to keep the modules that match what is in .fixtures.yml, or ends up pulling way too many modules down when SIMP_SPEC_PUPPETFILE is specified by itself.

Reply:
Well, the thought was that this would override anything in the local `.fixtures.yml`. Easy enough to map to a local fixtures file though.